xdsl/dialects/x86/ops.py

```diff
@@ -1195,6 +1191,65 @@ def custom_print_attributes(self, printer: Printer) -> Set[str]:
         return {"offset"}
 
 
+@irdl_op_definition
+class M_PopOp(IRDLOperation, X86Instruction, ABC):
+    """
+    Copies the value at the top of the stack into [r1] and increases %rsp.
+    https://www.felixcloutier.com/x86/pop
+    """
+
+    name = "x86.m.pop"
+
+    rsp_input = operand_def(GeneralRegisterType("rsp"))
+    destination = operand_def(
+        R1InvT
+    )  # not an operand, but couldn't think of a better way to make it part of the IR
```

Comment:
Can you please describe what's happening and what you're aming for a bit more?

Reply:
So the problem is that the value within that memory location doesn't matter so it's not really an operand. We did it in a similar way in the register version of pop. In this case however the destination is a memory access, not a register so no SSA value is created either. As such, I didn't see any way to include the information of what the memory access actually is in the IR because it doesn't fit neither operand_def nor result_def.

Reply:
The register holds a pointer, not a value, right? My understanding right now is that an operand is clearly the way to model this, although maybe I'm missing something. Could you maybe explain the relevant values in registers and memory before and after this instruction is executed? The link in the docstring is not very clear.